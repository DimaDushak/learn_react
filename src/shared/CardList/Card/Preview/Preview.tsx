/* eslint-disable jsx-a11y/img-redundant-alt */
import React, { useContext } from 'react';
import postContext from '../../../context/postContext';
import styles from './preview.css';

export default function Preview(): JSX.Element {
    const post = useContext(postContext);

    const imgSrc = post.preview?.images[0].source.url;

    return (
        <>
            {imgSrc && (
                <div className={styles.preview}>
                    <img
                        className={styles.previewImage}
                        alt="Post picture"
                        src={imgSrc.replaceAll('&amp;', '&')}
                    />
                </div>
            )}
        </>
    );
}
