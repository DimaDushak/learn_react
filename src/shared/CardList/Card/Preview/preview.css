.preview {
    max-height: 178px;
}

.previewImage {
    height: 100%;
    object-fit: cover;
    width: 100%;
}

@media (min-width: 1024px) {
    .preview {
        max-height: 107px;
        max-width: 190px;
        order: -1;
        overflow: hidden;
    }
}
