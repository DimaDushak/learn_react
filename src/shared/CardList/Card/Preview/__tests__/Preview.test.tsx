import { render } from '@testing-library/react';
import React from 'react';
import Preview from '../Preview';

test('should render Preview (snapshot)', () => {
    const { container } = render(<Preview />);

    expect(container.firstChild).toMatchSnapshot();
});
