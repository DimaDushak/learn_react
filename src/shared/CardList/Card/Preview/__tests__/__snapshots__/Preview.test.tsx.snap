// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Preview (snapshot) 1`] = `
<div
  class="preview"
>
  <img
    alt="#"
    class="previewImage"
    src="/static/fonts/Rectangle.png"
  />
</div>
`;
